src/openforms/submissions/form_logic.py

```diff
@@ -188,29 +190,17 @@ def evaluate_form_logic(
         for component in config_wrapper:
             key = component["key"]
             # already processed, don't process it again
-            if glom(data_diff, key, default=empty) is not empty:
+            if data_diff.get(key, default=empty) is not empty:
                 continue
 
-            new_value = glom(updated_step_data, key, default=empty)
-            original_value = glom(initial_data, key, default=empty)
-            # Reset the value of any field that may have become hidden again after evaluating the logic
-            if original_value is not empty and original_value != (
-                component_empty_value := get_component_empty_value(component)
-            ):
-                if (
-                    component
-                    and not is_visible_in_frontend(component, data_container.data)
-                    and component.get("clearOnHide")
-                ):
-                    assign(data_diff, key, component_empty_value, missing=dict)
-                    continue
-
```

Comment:
I'm confused by how it still works :thinking: Where do we check the `clearOnHide` now?

Reply:
it's a couple lines above actually! close the initialization of `data_diff`. The exact same code but written differently.

I tried adding a test for the missing coverage, but then it turned out that this line of code was just never hit (dead code) because of the code above. CTRL + F on `clearOnHide` and you'll find it.